src/internal/operators/withLatestFrom.ts

```diff
@@ -1,107 +1,24 @@
-  v4: O4,
-  v5: O5,
-  project: (v1: T, v2: ObservedValueOf<O2>, v3: ObservedValueOf<O3>, v4: ObservedValueOf<O4>, v5: ObservedValueOf<O5>) => R
-): OperatorFunction<T, R>;
-export function withLatestFrom<
-  T,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>,
-  O6 extends ObservableInput<any>,
-  R
->(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5,
-  v6: O6,
-  project: (
-    v1: T,
-    v2: ObservedValueOf<O2>,
-    v3: ObservedValueOf<O3>,
-    v4: ObservedValueOf<O4>,
-    v5: ObservedValueOf<O5>,
-    v6: ObservedValueOf<O6>
-  ) => R
+export function withLatestFrom<T, O extends unknown[]>(inputs: [...ObservableInputTuple<O>]): OperatorFunction<T, [T, ...O]>;
+
+export function withLatestFrom<T, O extends unknown[]>(...inputs: [...ObservableInputTuple<O>]): OperatorFunction<T, [T, ...O]>;
+
+export function withLatestFrom<T, O extends unknown[], R>(
+  inputs: [...ObservableInputTuple<O>],
+  project: (...value: [T, ...O]) => R
 ): OperatorFunction<T, R>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>>(source2: O2): OperatorFunction<T, [T, ObservedValueOf<O2>]>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>>(
-  v2: O2,
-  v3: O3
-): OperatorFunction<T, [T, ObservedValueOf<O2>, ObservedValueOf<O3>]>;
-export function withLatestFrom<T, O2 extends ObservableInput<any>, O3 extends ObservableInput<any>, O4 extends ObservableInput<any>>(
-  v2: O2,
-  v3: O3,
-  v4: O4
-): OperatorFunction<T, [T, ObservedValueOf<O2>, ObservedValueOf<O3>, ObservedValueOf<O4>]>;
-export function withLatestFrom<
-  T,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>
->(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5
-): OperatorFunction<T, [T, ObservedValueOf<O2>, ObservedValueOf<O3>, ObservedValueOf<O4>, ObservedValueOf<O5>]>;
-export function withLatestFrom<
-  T,
-  O2 extends ObservableInput<any>,
-  O3 extends ObservableInput<any>,
-  O4 extends ObservableInput<any>,
-  O5 extends ObservableInput<any>,
-  O6 extends ObservableInput<any>
->(
-  v2: O2,
-  v3: O3,
-  v4: O4,
-  v5: O5,
-  v6: O6
-): OperatorFunction<T, [T, ObservedValueOf<O2>, ObservedValueOf<O3>, ObservedValueOf<O4>, ObservedValueOf<O5>, ObservedValueOf<O6>]>;
-export function withLatestFrom<T, R>(...observables: Array<ObservableInput<any> | ((...values: Array<any>) => R)>): OperatorFunction<T, R>;
-export function withLatestFrom<T, R>(array: ObservableInput<any>[]): OperatorFunction<T, R>;
-export function withLatestFrom<T, R>(array: ObservableInput<any>[], project: (...values: Array<any>) => R): OperatorFunction<T, R>;
 
-/* tslint:enable:max-line-length */
+export function withLatestFrom<T, O extends ObservableInput<unknown>[], R>(
+  ...inputs: [...O, (...value: [T, ...ObservedValueTupleFromArray<O>]) => R]
+): OperatorFunction<T, R>;
```

Comment:
issue: It should be possible to delete this. After the N-args refactors, we don't want any type parameter constraints to be using `ObservableInput`. That type should be used only for selector return types.

Reply:
Yep, I refactored the other signatures so this is no longer necessary.